Cython/Compiler/ParseTreeTransforms.py

```diff
@@ -4152,12 +4152,36 @@ def visit_SimpleCallNode(self, node):
                     error(node.function.pos,
                           "cast() takes exactly two arguments and an optional typecheck keyword")
                 else:
-                    type = node.args[0].analyse_as_type(self.current_env())
-                    if type:
-                        node = ExprNodes.TypecastNode(
-                            node.function.pos, type=type, operand=node.args[1], typecheck=False)
+                    if isinstance(node.args[0], (ExprNodes.IndexNode, ExprNodes.SliceIndexNode)):
+                        if hasattr(node.args[0], 'index') and node.args[0].index.start.is_none:
```

Comment:
I think it'd be better to have separate paths for `IndexNode` and `SliceIndexNode` instead of using `hasattr`

Reply:
That, and `IndexNode` can be detected with `node.is_subscript`. `SliceIndexNode` can't, though.